Amethyst/Layout/ThreeColumnLayout.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  ThreeColumnLayout.swift
+//  Amethyst
+//
+//  Originally created by Ian Ynda-Hummel on 12/15/15.
+//  Copyright © 2015 Ian Ynda-Hummel. All rights reserved.
+//
+//  Modifications by Craig Disselkoen on 08/31/18.
+//
+
+import Silica
+
+enum WhichColumn {
+  case left
+  case middle
+  case right
+}
+
+enum WhichPane {
+  case main
+  case secondary
+  case tertiary
+}
+
+final class ThreeColumnReflowOperation: ReflowOperation {
+    private let layout: ThreeColumnLayout
+    private let mainIs: WhichColumn
+
+    init(screen: NSScreen, windows: [SIWindow], layout: ThreeColumnLayout, frameAssigner: FrameAssigner, mainIs: WhichColumn) {
+        self.layout = layout
+        self.mainIs = mainIs
+        super.init(screen: screen, windows: windows, frameAssigner: frameAssigner)
+    }
+
+    func frameAssignments() -> [FrameAssignment] {
```

Comment:
Tried to clean up a bit.  Just pushed a new version that has complexity 29 instead of 35.  There appears to be a lot going on in this function, but that is probably because so many intermediate results are named and assigned to variables.  I think this increases clarity rather than reducing it.